src/sys/signal.rs

```diff
@@ -9,11 +9,37 @@ use std::mem;
 #[cfg(any(target_os = "dragonfly", target_os = "freebsd"))]
 use std::os::unix::io::RawFd;
 use std::ptr;
+use std::ops::{BitOr, BitOrAssign};
 
 // Currently there is only one definition of c_int in libc, as well as only one
 // type for signal constants.
 // We would prefer to use the libc::c_int alias in the repr attribute. Unfortunately
 // this is not (yet) possible.
+/// Signal is a representation of a POSIX signal on various platforms, analogous to the libc::SIG*
+/// constants. In nix it is used for various operations related to signals (such as masking,
+/// sending, intercepting, etc). A related type is SigSet, which represents a collection (or set)
+/// of Signals.
+///
+/// # Examples
+///
+/// Note that any Signal can be promoted to a SigSet either through the use of the `|` operator, or
+/// through std::convert::Into.
+///
+/// ```
+/// # use ::nix::sys::signal::*;
+/// // Into and From promote a signal to a SigSet that has only the given signal inside it.
```

Comment:
Please leave these comments out of the code example instead and separate these into two separate examples. It'll make it slightly easier reading. Something like:
```
/// Into and From promote a signal to a SigSet that has only the given signal inside it.
/// ```
/// # use ::nix::sys::signal::*;
/// let sigset_a: SigSet = Signal::SIGTERM.into();
/// let sigset_b = SigSet::from(Signal::SIGTERM);
///```
///
/// `SigSet`s of multiple signals can be created by using the `|` (`BitOr`) operator:
/// ```
/// # use ::nix::sys::signal::*;
/// let sigset = Signal::SIGTERM | Signal::SIGHUP;
/// ```
///
/// Existing `SigSet`s can be extended using the `|` (`BitOr`) or `|=` (`BitAssign`) operator as well:
/// ```
/// let mut sigset = SigSet::empty();
/// sigset |= Signal::SIGQUIT;
/// ```
```